internal/service/kms/secrets_data_source.go

```diff
@@ -21,24 +22,33 @@ func DataSourceSecrets() *schema.Resource {
 				Required: true,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
-						"name": {
-							Type:     schema.TypeString,
-							Required: true,
-						},
-						"payload": {
-							Type:     schema.TypeString,
-							Required: true,
-						},
 						"context": {
 							Type:     schema.TypeMap,
 							Optional: true,
 							Elem:     &schema.Schema{Type: schema.TypeString},
 						},
+						"encryption_algorithm": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringInSlice(kms.EncryptionAlgorithmSpec_Values(), false),
+						},
 						"grant_tokens": {
 							Type:     schema.TypeList,
 							Optional: true,
 							Elem:     &schema.Schema{Type: schema.TypeString},
 						},
+						"keyid": {
```

Comment:
Since the documentation at [KMS KeyId](https://docs.aws.amazon.com/kms/latest/APIReference/API_Encrypt.html#KMS-Encrypt-request-KeyId) uses `KeyId`, I think we can refactor this to use

```suggestion
						"key_id": {

Reply:
Done as Requested